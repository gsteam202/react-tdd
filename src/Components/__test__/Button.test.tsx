import { fireEvent, render, screen } from '@testing-library/react';
import 'jest-styled-components';

import { Button } from '../';

describe('<Button />', () => {
  it('정확하게 렌더링 되는지 테스트', () => {
    const { container } = render(<Button label="Button Test" />);

    const label = screen.getByText('Button Test');

    expect(label).toBeInTheDocument(); // 문서에 존재하는지 확인

    const parent = label.parentElement; //

    expect(parent).toHaveStyleRule('background-color', '#304ffe');
    expect(parent).toHaveStyleRule('background-color', '#1e40ff', {
      modifier: ':hover',
    });

    expect(container).toMatchSnapshot();
  });

  it('backgroundColor, hoverColor props test', () => {
    const backgroundColor = '#ff1744';
    const hoverColor = '#f01440';

    render(
      <Button
        label="Button Test"
        backgroundColor={backgroundColor}
        hoverColor={hoverColor}
      />
    );

    const parent = screen.getByText('Button Test').parentElement;
    expect(parent).toHaveStyleRule('background-color', backgroundColor);
    expect(parent).toHaveStyleRule('background-color', hoverColor, {
      modifier: ':hover',
    });
  });

  it('클릭이벤트 테스트', () => {
    const handleClick = jest.fn();
    render(<Button label="Button Test" onClick={handleClick} />);

    const label = screen.getByText('Button Test');
    expect(handleClick).toHaveBeenCalledTimes(0);
    fireEvent.click(label);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
