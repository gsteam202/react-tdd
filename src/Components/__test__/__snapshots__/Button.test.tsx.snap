// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> 정확하게 렌더링 되는지 테스트 1`] = `
.c0 {
  text-align: center;
  font-weight: bold;
  background-color: #304ffe;
  padding: 10px 20px;
  border-radius: 8px;
  border: 0;
}

.c0:hover {
  background-color: #1e40ff;
}

.c0:active {
  box-shadow: insect 5px 5px 10px rgba(0,0,0,0.1);
}

.c1 {
  color: #fff;
  font-size: 16px;
}

<div>
  <button
    class="c0"
  >
    <div
      class="c1"
    >
      Button Test
    </div>
  </button>
</div>
`;
